File: v1/src/main/java/com/api/v1/mappers/payment/PaymentResponseMapper.java
package com.api.v1.mappers.payment;

import com.api.v1.domain.creditcard.CreditCard;
import com.api.v1.domain.payment.Payment;
import com.api.v1.dtos.payment.PaymentResponseDto;
import com.api.v1.mappers.creditcard.CreditCardResponseMapper;

public record PaymentResponseMapper(
        Payment payment,
        CreditCard creditCard
) {

    public PaymentResponseDto map() {
        return new PaymentResponseDto(
                payment.getOrderNumber(),
                new CreditCardResponseMapper(creditCard).mapToDto(),
                payment.getPaidAt()
        );
    }

}

File: kotlin/v2/src/main/kotlin/com/api/v2/payment/mappers/PaymentResponseMapper.kt
package com.api.v2.payment.mappers

import com.api.v2.creditcard.domain.CreditCard
import com.api.v2.creditcard.mappers.CreditCardResponseMapper
import com.api.v2.payment.domain.Payment
import com.api.v2.payment.dtos.PaymentResponseDto

class PaymentResponseMapper {

    companion object {
        fun map(payment: Payment, creditCard: CreditCard): PaymentResponseDto {
            return PaymentResponseDto(
                payment.orderNumber,
                CreditCardResponseMapper.mapToDto(creditCard),
                payment.paidAt
            )
        }
    }

}